config/Make.rules.Linux

```diff
@@ -1,31 +1,17 @@
 # Copyright (c) ZeroC, Inc.
 
-known_distributions             := centos rhel fedora debian ubuntu amzn sles
-linux_ids                       := $(if $(wildcard /etc/os-release),$(shell . /etc/os-release && echo $${ID} $${ID_LIKE}),)
-linux_id                        ?= $(word 1,$(or $(filter $(known_distributions),$(linux_ids)),$(linux_ids)))
-is-bin-program                  = $(and $(filter $(bindir)%,$($4_targetdir)),$(filter $($1_target),program))
-
-ifneq ($(OECORE_SDK_VERSION),)
-linux_id                       = yocto
-endif
-
-ifneq ($(filter yocto poky,$(linux_id)),)
+known_distributions             := centos rhel fedora debian ubuntu
 
-build-platform                 = default
-supported-platforms            = $(build-platform)
+# Assign the linux_ids from the ID and ID_LIKE variables in /etc/os-release if available.
+linux_ids                       := $(if $(wildcard /etc/os-release),$(shell . /etc/os-release && echo $${ID} $${ID_LIKE}),)
 
-$(build-platform)_cc           = $(CC)
-$(build-platform)_cxx          = $(CXX)
-$(build-platform)_cppflags     = -Wno-psabi
-$(build-platform)_libtool      = $(AR)
-$(build-platform)_targetdir    = $(if $(filter %/build,$5),/$(build-platform))
-$(build-platform)_installdir   =
-$(build-platform)_objdir       =
+# Assign the linux_id to the first known distribution in the linux_ids list, or for unknown distributions to the first
+# element in the linux_ids list.
+linux_id                        ?= $(firstword $(or $(filter $(known_distributions),$(linux_ids)),$(linux_ids)))
 
-# system install directory
-system-install-dir              := $(OECORE_TARGET_SYSROOT)/usr
```

Comment:
Are these no longer needed?

Reply:
This was for Yocto, if we ever want to add back support for it we can bring it back. But so far we are not testing/supporting this with 3.8.